classes/citation/Repository.php

```diff
@@ -0,0 +1,288 @@
+                        $citation->setData('isProcessed', false);
+                        $citationId = $this->dao->insert($citation);
+                        $importedCitations[] = $citation;
+                        $this->addJobForCitation($citationId);
+                    }
+                }
+            }
+
+            Hook::call('Citation::importCitations::after', [$publicationId, $existingCitations, $importedCitations]);
+        }
+    }
+
+    /**
+     * Import citations from a raw citation list of the particular publication to existing citations.
+     */
+    public function importAdditionalCitations(int $publicationId, ?string $rawCitationList): string
+    {
+        $citationTokenizer = new CitationListTokenizerFilter();
+        $citationStrings = $rawCitationList ? $citationTokenizer->execute($rawCitationList) : [];
+        $existingSeq = collect($this->getByPublicationId($publicationId))
+            ->map(fn(Citation $citation) => $citation->getData('seq'))
+            ->all();
+        $lastSeq = end($existingSeq) ? end($existingSeq) + 1 : 1;
+
+        $rejectedCitations = [];
+        if (is_array($citationStrings) && !empty($citationStrings)) {
+            foreach ($citationStrings as $rawCitationString) {
+                if (!empty(trim($rawCitationString))) {
+                    if (!$this->existsRawCitation($publicationId, $rawCitationString)) {
+                        $citation = new Citation();
+                        $citation->setRawCitation($rawCitationString);
+                        $citation->setData('publicationId', $publicationId);
+                        $citation->setSequence($lastSeq);
+                        $citation->setData('isProcessed', false);
+                        $newCitationId = $this->dao->insert($citation);
+                        $this->addJobForCitation($newCitationId);
+                        $lastSeq++;
+                    } else {
+                        $rejectedCitations[] = $rawCitationString;
+                    }
+                }
+            }
+        }
+
+        return implode(PHP_EOL, $rejectedCitations);
+    }
+
+    /** @copydoc DAO::existsRawCitation() */
+    public function existsRawCitation(int $publicationId, string $rawCitation): bool
+    {
+        return $this->dao->existsRawCitation($publicationId, $rawCitation);
+    }
+
+    /**
+     * Add a new job for metadata lookup for a given publication.
+     */
+    public function addJobForPublication(int $publicationId): void
+    {
+        $citations = $this->getByPublicationId($publicationId);
+
+        if (empty($citations)) {
+            return;
+        }
+
+        foreach ($citations as $citation) {
+            $this->addJobForCitation($citation->getId());
+        }
+    }
+
+    /**
+     * Add a new job for metadata lookup for a given publication.
+     */
+    public function addJobForCitation(int $citationId, int $jobDelay = 60): void
```

Comment:
Maybe we do not need this function, the job can be directly dispatched. It is only called when importing a citation, that is not processed in that moment.